diff --git a/src/jwt/src/claims.rs b/src/jwt/src/claims.rs
index 5f40eec..1432442 100644
--- a/src/jwt/src/claims.rs
+++ b/src/jwt/src/claims.rs
@@ -174,6 +174,8 @@ pub struct JwtIdClaims<'a> {
     pub groups: Option<Vec<String>>, // TODO change to borrowed data when everything works
     #[serde(skip_serializing_if = "Option::is_none")]
     pub custom: Option<HashMap<String, serde_json::Value>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub vikunja_groups: Option<serde_json::Value>,
     #[serde(borrow, skip_serializing_if = "Option::is_none")]
     pub webid: Option<Cow<'a, str>>,
     #[serde(borrow, skip_serializing_if = "Option::is_none")]
diff --git a/src/service/src/token_set.rs b/src/service/src/token_set.rs
index 58be770..9ca6f6b 100644
--- a/src/service/src/token_set.rs
+++ b/src/service/src/token_set.rs
@@ -295,6 +295,7 @@ impl TokenSet {
             roles: user.get_roles(),
             groups: None,
             custom: None,
+            vikunja_groups: None,
             webid,
             zoneinfo: None,
         };
@@ -363,7 +364,10 @@ impl TokenSet {
                 }
             }
             if !attr.is_empty() {
-                claims.custom = Some(attr);
+                claims.vikunja_groups = attr.remove("vikunja_groups");
+                if !attr.is_empty() {
+                    claims.custom = Some(attr);
+                }
             }
         }
 
